notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  *\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "druid.display.run_task(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1776cd84-9f38-40a3-9e47-5e821d15c901",
+   "metadata": {},
+   "source": [
+    "`REPLACE` or `INSERT` at the beginning of the statement tells Druid to execute an ingestion task. `INSERT` is used when appending data, `REPLACE` when replacing data. Both methods work when adding data to a new or empty Druid datasource. The `OVERWRITE ALL` clause means that the whole datasource will be replaced with the result of this ingestion. \n",
+    "\n",
+    "```\n",
+    "REPLACE INTO \"wikipedia_events\" OVERWRITE ALL\n",
+    "```\n",
+    "\n",
+    "The `WITH` clause is used to declare one or more input sources, this could also be placed directly in the `FROM` clause of the final `SELECT`, but this is easier to read:\n",
+    "\n",
+    "```\n",
+    "WITH \"ext\" AS \n",
+    "(\n",
+    "    SELECT ...\n",
+    "      FROM TABLE (EXTERN ( ... ) )\n",
+    ") EXTEND (...)\n",
+    "```\n",
+    "\n",
+    "`EXTERN` supports many batch [input sources](https://druid.apache.org/docs/latest/ingestion/native-batch-input-sources.html) and [formats](https://druid.apache.org/docs/latest/ingestion/data-formats.html). In this case we are using input source type `http` to access a set or `uris` that each contain a data file in the `json` data format. Note that compressed files are allowed and will automatically be decompressed.\n",
+    "```\n",
+    "FROM TABLE(\n",
+    "      EXTERN(\n",
+    "        '{\"type\":\"http\",\"uris\":[\"https://druid.apache.org/data/wikipedia.json.gz\"]}',\n",
+    "        '{\"type\":\"json\"}'\n",
+    "      )\n",
+    "```\n",
+    "The `EXTEND` clause describes the input schema using SQL data types:\n",
+    "```\n",
+    "EXTEND (\"isRobot\" VARCHAR, \"channel\" VARCHAR, \"timestamp\" VARCHAR,  ...)\n",
+    "```\n",
+    "\n",
+    "The final SELECT statement defines the transformations and schema of the resulting Druid table. A `__time` column is usually parsed from the source, this expression will be mapped to Druid's primary time partitioning of segments. In this case we specified the `__time` column and ingested the rest of the columns \"AS IS\" using `*`.\n",
+    "\n",
+    "```\n",
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  *\n",
+    "FROM \"ext\"\n",
+    "```\n",
+    "\n",
+    "The final portion of this ingestion is the `PARTITIONED BY DAY` clause which tells Driud to create a separate set of segments for each day. A `PARTITION BY` clause must be included in all `INSERT`/`REPLACE` statements."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "66b30164-b3c3-4de6-ba68-d269f0bee2d1",
+   "metadata": {},
+   "source": [
+    "#### Wait for Segment Availibility\n",
```

Comment:
As people will take some time to read your commentary, I reckon you could just put this at the end of the cell with the ingestion in it, else it doesn't really do anything.

Reply:
done